### ISSUE TITLE
Conv1D -> Flatten Not Working in Keras Loader (1.0.0-alpha)

### ISSUE THREAD

**Issue by mobiusinversion** [deeplearning4j/deeplearning4j#5061]
Hello,

I have trained and stored a Keras model, but I cannot import it into DL4J.

    [INFO] 2018-05-05 09:59:36,540 [Reflections:229] Reflections took 5985 ms to scan 168 urls, producing 3656 keys and 15809 values 
    [INFO] 2018-05-05 09:59:37,070 [MultiLayerNetwork:55] Starting MultiLayerNetwork with WorkspaceModes set to [training: SEPARATE; inference: SEPARATE]

    java.lang.IllegalStateException: Mis matched lengths: [2048] != [32768] - Array 1 shape: [16, 128], array 2 shape: [256, 128]

	at org.nd4j.linalg.util.LinAlgExceptions.assertSameLength(LinAlgExceptions.java:42)
	at org.nd4j.linalg.api.ops.BaseTransformOp.<init>(BaseTransformOp.java:174)
	at org.nd4j.linalg.api.ops.impl.transforms.Set.<init>(Set.java:43)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.assign(BaseNDArray.java:1274)
	at org.deeplearning4j.nn.layers.BaseLayer.setParam(BaseLayer.java:204)

My Keras model is structured as follows:

    model = Sequential()
    model.add(Embedding(len(word_index) + 1, embedding_size,
                    weights=[embedding_matrix],
                    input_length=MAX_SEQUENCE_LENGTH,
                    trainable=True))

    model.add(Conv1D(128, 5, activation='relu'))
    model.add(Conv1D(64, 5, activation='relu'))
    model.add(Conv1D(32, 5, activation='relu'))
    model.add(Conv1D(16, 5, activation='relu'))
    model.add(Flatten())
    model.add(Dense(128, activation='relu'))
    model.add(Dropout(0.2))
    model.add(Dense(Y.shape[1], activation='softmax'))

One thing that is not immediately clear is where the 256 comes from above in Array 2 shape.  But it does appear to be an issue between the last convolutional layer and the flattening.

**Comment by maxpumperla** [comment]
I'll look into Flatten and Reshape next week and add you example as unit test. thanks @mobiusinversion

**Comment by maxpumperla** [comment]
@mobiusinversion we have a complete e2e test for your use case, works on master and should also work on snapshots already. this has been around for a while, so I'm not exactly sure why this happens for you. See here:

https://github.com/deeplearning4j/deeplearning4j/blob/4f2079cfcc86dd5b74a9cfcf0ea022691969a471/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/weights/scripts/embedding_conv1d_extended.py#L24-L32

https://github.com/deeplearning4j/deeplearning4j/blob/4f2079cfcc86dd5b74a9cfcf0ea022691969a471/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/weights/KerasWeightSettingTests.java#L52-L56
